seatunnel-e2e/seatunnel-connector-v2-e2e/connector-hbase-e2e/src/test/java/org/apache/seatunnel/e2e/connector/hbase/HbaseIT.java

```diff
@@ -109,59 +83,91 @@ public void tearDown() throws Exception {
         if (Objects.nonNull(admin)) {
             admin.close();
         }
-        if (Objects.nonNull(hbaseConnection)) {
-            hbaseConnection.close();
-        }
-        if (Objects.nonNull(hbaseContainer)) {
-            hbaseContainer.close();
-        }
+        hbaseCluster.stopService();
     }
 
     private void initialize() throws IOException {
-        hbaseConfiguration.set("hbase.zookeeper.quorum", HOST + ":" + PORT);
-        hbaseConnection = ConnectionFactory.createConnection(hbaseConfiguration);
-        admin = hbaseConnection.getAdmin();
-        table = TableName.valueOf(TABLE_NAME);
-        ColumnFamilyDescriptor familyDescriptor =
-                ColumnFamilyDescriptorBuilder.newBuilder(FAMILY_NAME.getBytes())
-                        .setCompressionType(Compression.Algorithm.SNAPPY)
-                        .setCompactionCompressionType(Compression.Algorithm.SNAPPY)
-                        .build();
-        TableDescriptor tableDescriptor =
-                TableDescriptorBuilder.newBuilder(table).setColumnFamily(familyDescriptor).build();
-        admin.createTable(tableDescriptor);
-        log.info("Hbase table has been initialized");
+        // Create table for hbase sink test
+        System.out.println("initial");
+        hbaseCluster.createTable(TABLE_NAME, Arrays.asList(FAMILY_NAME));
+        // Create table and insert data for hbase source test
+        hbaseCluster.createTable("test_table", Arrays.asList("cf1", "cf2"));
+        hbaseCluster.putRow("test_table", "row1", "cf1", "col1", "True"); // boolean
+        hbaseCluster.putRow("test_table", "row1", "cf1", "col2", "3.923"); // double
+        hbaseCluster.putRow("test_table", "row1", "cf2", "col1", "4537654375638"); // bigint
+        hbaseCluster.putRow("test_table", "row1", "cf2", "col2", "33"); // int
+        hbaseCluster.putRow("test_table", "row1", "cf2", "col_date", "2024-08-12"); // date
+        hbaseCluster.putRow(
+                "test_table", "row1", "cf2", "col_timestamp", "2024-08-27 03:15:10"); // timestamp
+        hbaseCluster.putRow("test_table", "row1", "cf2", "col_time", "04:44:09"); // time
+        hbaseCluster.putRow("test_table", "row2", "cf1", "col1", "False"); // boolean
+        hbaseCluster.putRow("test_table", "row2", "cf1", "col2", "465.573"); // double
+        hbaseCluster.putRow("test_table", "row2", "cf2", "col1", "7893658245220"); // bigint
+        hbaseCluster.putRow("test_table", "row2", "cf2", "col2", "31"); // int
+        hbaseCluster.putRow("test_table", "row2", "cf2", "col_date", "2024-04-03"); // date
+        hbaseCluster.putRow(
+                "test_table", "row2", "cf2", "col_timestamp", "2024-11-22 10:41:49"); // timestamp
+        hbaseCluster.putRow("test_table", "row2", "cf2", "col_time", "14:24:38"); // time
+        System.out.println("Hbase table has been initialized");
```

Comment:
**Scalability Issue**: The code initializes and populates HBase tables synchronously within the test initialization method. This approach can lead to scalability issues when dealing with large datasets or high-frequency test executions, as it does not utilize asynchronous or batch processing techniques, potentially causing delays and performance bottlenecks. <br> **Fix**: Implement asynchronous data insertion and batch processing for initializing and populating HBase tables to enhance scalability and performance. Utilize HBase's batch processing capabilities to insert multiple rows simultaneously. <br> **Code Suggestion**: 
 ```
 +        // Asynchronously create table and insert data for scalability
 +        CompletableFuture<Void> setupFuture = CompletableFuture.runAsync(() -> {
 +            try {
 +                hbaseCluster.createTable(TABLE_NAME, Arrays.asList(FAMILY_NAME));
 +                // Batch insert example
 +                List<Put> puts = new ArrayList<>();
 +                puts.add(new Put(Bytes.toBytes("row1")).addColumn(Bytes.toBytes("cf1"), Bytes.toBytes("col1"), Bytes.toBytes("True")));
 +                puts.add(new Put(Bytes.toBytes("row1")).addColumn(Bytes.toBytes("cf1"), Bytes.toBytes("col2"), Bytes.toBytes("3.923")));
 +                // Add more puts as needed
 +                hbaseCluster.getTable(TABLE_NAME).put(puts);
 +                System.out.println("Hbase table has been initialized asynchronously");
 +            } catch (IOException e) {
 +                e.printStackTrace();
 +            }
 +        });
 +        setupFuture.join(); // Ensure completion before proceeding with tests
 ```